torch/autograd/profiler_util.py

```diff
@@ -562,21 +630,23 @@ def cuda_time_total(self):
             else:
                 # each legacy cpu events has a single (fake) kernel
                 return sum(kinfo.duration for kinfo in self.kernels)
-        else:
-            assert self.device_type == DeviceType.CUDA
+        elif self.device_type == DeviceType.CUDA:
             return self.time_range.elapsed_us()
+        else:
+            return 0
 
     @property
     def self_cuda_time_total(self):
-        if self.is_async or self.use_device:
+        if self.is_async:
             return 0
         if self.device_type == DeviceType.CPU:
             return self.cuda_time_total - sum(
                 [child.cuda_time_total for child in self.cpu_children]
             )
+        elif self.device_type == DeviceType.CUDA:
+            return self.time_range.elapsed_us()
```

Comment:
I believe this should be `return self.cuda_time_total`

Reply:
Indeed, when the condition `self.device_type == DeviceType.CUDA` holds true, the values returned by `self.time_range.elapsed_us()` and `self.cuda_time_total` are functionally identical. This is because the latter is derived directly from the former. Within the implementation of the `cuda_time_total()` method, you can observe a branch where `self.device_type == DeviceType.CUDA` is checked, and it is in this branch that `self.time_range.elapsed_us()` is returned. However, opting to retrieve `cuda_time_total` would necessitate invoking the `cuda_time_total()` method, which includes additional time-consuming conditional checks. In contrast, calculating the duration directly from the `Interval` type's `time_range` involves only a simple arithmetic subtraction. Therefore, I would advocate for the use of `return self.time_range.elapsed_us()` for its simplicity and directness.

Reply:
Okay, does it make sense to use `self.time_range.elapsed_us()` for `self_xpu_time_total` and `self_privateuse1_time_total` as well? Since this code path is starting to diverge into 3, I want to make sure we are consistent across them.

Reply:
Yes, you're correct. It was an oversight on my part during the modifications. I have now updated the code to use a uniform approach, with `self.time_range.elapsed_us()` as the expression for calculating the time across all corresponding paths.